beacon-chain/chaintest/backend/helpers.go

```diff
@@ -37,6 +48,65 @@ func generateSimulatedBlock(
 			Exits:             []*pb.Exit{},
 		},
 	}
+	if simulatedDeposit != nil {
+		depositInput := &pb.DepositInput{
+			Pubkey:                      []byte(simulatedDeposit.Pubkey),
+			WithdrawalCredentialsHash32: []byte{},
+			ProofOfPossession:           []byte{},
+			RandaoCommitmentHash32:      depositRandaoCommit[:],
+			CustodyCommitmentHash32:     []byte{},
+		}
+		wBuf := new(bytes.Buffer)
+		if err := ssz.Encode(wBuf, depositInput); err != nil {
+			return nil, [32]byte{}, fmt.Errorf("failed to encode deposit input: %v", err)
+		}
+		encodedInput := wBuf.Bytes()
+		data := []byte{}
+
+		// We set a deposit value of 1000.
+		value := make([]byte, 8)
+		binary.BigEndian.PutUint64(value, simulatedDeposit.Amount)
+
+		// We then serialize a unix time into the timestamp []byte slice
+		// and ensure it has size of 8 bytes.
+		timestamp := make([]byte, 8)
+
+		// Set deposit time to 1000 seconds since unix time 0.
+		depositTime := time.Now().Unix()
+		binary.BigEndian.PutUint64(timestamp, uint64(depositTime))
+
+		// We then create a serialized deposit data slice of type []byte
+		// by appending all 3 items above together.
+		data = append(data, encodedInput...)
```

Comment:
I think you can do these three things in one append method.

```go
data = append(data, encodedInput..., value..., timestamp...)
```

It could be less copying? I'm not sure if it makes a performance difference.